### PR TITLE
Adds explanations about the entity user provider custom query behavior

### PR DESCRIPTION
**Pull request by xavismeh**
Follow up from https://github.com/symfony/symfony/issues/46187 discussion

### CODE REVIEW

File: security/user_provider.rst [hunk 1]
```diff
@@ -209,6 +209,17 @@ needed (e.g. because :doc:`user impersonation </security/impersonating_user>`,
 :doc:`Remember Me </security/remember_me>`, or some other security feature is
 activated), the ``loadUserByUsername()`` method on ``UserRepository`` will be called.
 
+.. caution::
+
+    This custom query will only be used when authenticating the user. When reloading
+    the user from session, it will internally relay on the ``refreshUser()`` method
```
Comment (OskarStark) [inline security/user_provider.rst]:
```suggestion
    This custom query will only be used when authenticating the user. Reloading
    the user from the session, will internally relay on the ``refreshUser()`` method
```

File: security/user_provider.rst [hunk 2]
```diff
@@ -209,6 +209,17 @@ needed (e.g. because :doc:`user impersonation </security/impersonating_user>`,
 :doc:`Remember Me </security/remember_me>`, or some other security feature is
 activated), the ``loadUserByUsername()`` method on ``UserRepository`` will be called.
 
+.. caution::
+
+    This custom query will only be used when authenticating the user. Reloading
+    the user from the session will internally relay on the ``refreshUser()`` method
```
Comment (HeahDude) [inline security/user_provider.rst:215]:
```suggestion
    the user from the session will internally rely on the ``refreshUser()`` method
```

**Review by HeahDude — APPROVED** [review]
LGTM (should be retargeted to 5.4)

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
I noticed, that the whole page was deleted in 5.4, because of the rewrite of the security component. 
Feel free to create a new PR against 5.4.

Thank you

**Pull request closed without merging**